tests/dataframe/test_decimals.py

```diff
@@ -24,7 +24,7 @@ def test_decimal_parquet_roundtrip() -> None:
         df.write_parquet(dirname)
         df_readback = daft.read_parquet(dirname).collect()
 
-    assert str(df.to_pydict()["decimal128"]) == str(df_readback.to_pydict()["decimal128"])
+        assert str(df.to_pydict()["decimal128"]) == str(df_readback.to_pydict()["decimal128"])
```

Comment:
this was weird, if i didn't do the indent, it would fail with: 

````
../../daft/api_annotations.py:31: in _wrap
    return timed_method(*args, **kwargs)
../../daft/analytics.py:185: in tracked_method
    return method(*args, **kwargs)
../../daft/dataframe/dataframe.py:1316: in to_pydict
    return result.to_pydict()
../../daft/runners/partitioning.py:214: in to_pydict
    merged_partition = self._get_merged_vpartition()
../../daft/runners/pyrunner.py:52: in _get_merged_vpartition
    return MicroPartition.concat([part for id, part in ids_and_partitions])

@classmethod
    def concat(cls, to_merge: list[MicroPartition]) -> MicroPartition:
        micropartitions = []
        for t in to_merge:
            if not isinstance(t, MicroPartition):
                raise TypeError(f"Expected a MicroPartition for concat, got {type(t)}")
            micropartitions.append(t._micropartition)
>       return MicroPartition._from_pymicropartition(_PyMicroPartition.concat(micropartitions))
E       ValueError: DaftError::External Internal IO Error when Opening: /var/folders/nl/4tv7_psn7fq3n8sn6b4cpvw80000gn/T/tmputa6b7lc/ce958d3e-9fdc-48c3-9c8b-1917639037cd-0.parquet:
E       Details:
E       No such file or directory (os error 2)
```

Reply:
oh i see, .collect() creates an unloaded micropartition, so it didn't do any actual reading of the parquet data